### DISCUSSION TITLE
Milvus Backup

### DISCUSSION THREAD

**Discussion by mgoyal-adeptia** [milvus-io/milvus#30853] [Q&A and General discussion]
Hi,

How can I take backup of milvus databases. I have setup Milvus on AKS(Azurw Kubernetes Service) with minio storage(disk option).

-Manish

**Comment by yhmo** [comment]
The [milvus-backup](https://github.com/zilliztech/milvus-backup) is the recommended tool to do backup/restore.

1. Download the latest version zip to your local: https://github.com/zilliztech/milvus-backup/releases  (Mac OS uses milvus-backup_Darwin_xxx.tar.gz, Linux OS uses milvus-backup_Linux_xxx.tar.gz)
2. Extract the zip to a local path, assume the path name is `/XXX/milvus-backup_XXXX_xxx/`
3. Download [this yaml](https://github.com/zilliztech/milvus-backup/blob/main/configs/backup.yaml) to the `/XXX/milvus-backup_XXXX_xxx//configs/backup.yaml`
4. Edit the backup.yaml, set the address/port your milvus server into the milvus.address and milvus.port, if the milvus server enabled authentication, you need to set username and password
5. Edit the backup.yaml, set the minio including address, port, secretkey, accesskey, etc
6. In terminal, enter the path `/XXX/milvus-backup_XXXX_xxx/`
7. Start the milvus-backup server by command `./milvus-backup server`
8. Use curl command to do backup or restore: https://github.com/zilliztech/milvus-backup?tab=readme-ov-file#api-reference

**Comment by yhmo** [comment]
Attach a doc about how to use milvus-backup tool:
[milvus-backup tool practice.pdf](https://github.com/milvus-io/milvus/files/15320031/milvus-backup.tool.practice.pdf)
